CMS-Enterprise/sbom-harbor
# refactor: long-standing cleanup tasks

**NOTE TO REVIEWERS**: This is a PR around code organization and file/struct naming. No logic changes were made. You do not need to review implementation logic.

## Summary

This PR addresses several long standing cleanup tasks, primarily around code organization and naming. Because of the file moves and renaming, a very large number of files were affected.  To make it easier to review, the different refactoring tasks outlined below were done as separate commits but in the same PR. This should make reviewing each effort simpler for the reviewer. Separate PRs, would have made this effort take far too long amount of time. 

### Added

N/A

### Changed


#### CLI

* Changed the `harbor-cli` binary name to `harbor`.
* Renamed the `sbom` command to `ingest` so that it is a verb that matches our domain language.
* Created module directories for each command (ingest, enrich, analyze).
* Split up large mod.rs files in each module directory so we have more focused code files.

#### Directory Structure

* Moved the `core` crate to the `sdk` folder.
* Moved the `platform` crate to the `sdk` folder.
* Moved the `mongodb` module under the `persistence` module in the `platform` crate.
* Renamed `book` folder to `docs`.
* Updated paths in the documentation that reference `core` or `platform`.
* Updated the workspace and crate-specific Cargo.toml files to reflect the changes.
* Refactored any code that is affected by the module structure changes.

#### Core

* Moved the `TaskProvider` implementations to a top level module called `tasks`.
* Renamed the various `provider.rs` files to their respective task names.
* Made sure task names align with commands.
* Refactored any code that is affected by the module structure changes.
* 
### Deprecated

* Using structs (e.g. EnrichmentProvider) to handle CLI commands. Those structs predated the use of `clap` which has 
  it's own CommandFactory functionality essentially. Functions were simpler and sufficient given that feature of `clap`.
* Using a `new_provider` function to instantiate `TaskProvider` implementations. The `TaskProviders` all have their own 
  `new` functions and this was straight forward to inline.

### Removed

* Deleted the no-longer used `client` crate.
* Deleted the no-longer used `pilot` crate.
* Deleted any functions and tests I could find that weren't actually called by anything anymore.

### Fixed

N/A

## How to test

Pull the branch and run `cargo test` to make sure it all compiles and the tests work.  Each commit is actually runnable and was tested along the way.
